libs/iterator_support/CMakeLists.txt

```diff
@@ -33,13 +36,16 @@ add_hpx_module(iterator_support
   FORCE_LINKING_GEN
   GLOBAL_HEADER_GEN ON
   EXCLUDE_FROM_GLOBAL_HEADER
-    "hpx/iterator_support/is_iterator.hpp"
-    "hpx/iterator_support/is_range.hpp"
+    "hpx/iterator_support/traits/is_iterator.hpp"
+    "hpx/iterator_support/traits/is_range.hpp"
   SOURCES ${iterator_support_sources}
   HEADERS ${iterator_support_headers}
   COMPAT_HEADERS ${iterator_support_compat_headers}
   DEPENDENCIES
     hpx_config
+    hpx_datastructures
+    hpx_functional
+    hpx_serialization
```

Comment:
There's still what *seems* like a cycle (`iterator_support` - `serialization` - `util` - `iterator_support`) according to the dependencies, but the dependency checker isn't complaining. Either the checker is wrong, or there are unnecessary dependencies in this list.

Reply:
Thanks for noticing. The dependency of `serialization` on `util` was a leftover which should be gone now.